src/main/java/com/epam/test_generator/services/TestResultService.java

```diff
@@ -0,0 +1,290 @@
+package com.epam.test_generator.services;
+
+import com.epam.test_generator.config.security.AuthenticatedUser;
+import com.epam.test_generator.dao.interfaces.TestResultDAO;
+import com.epam.test_generator.dto.RawCaseResultDTO;
+import com.epam.test_generator.dto.RawStepResultDTO;
+import com.epam.test_generator.dto.RawSuitResultDTO;
+import com.epam.test_generator.dto.TestResultDTO;
+import com.epam.test_generator.entities.Case;
+import com.epam.test_generator.entities.CaseResult;
+import com.epam.test_generator.entities.Project;
+import com.epam.test_generator.entities.Status;
+import com.epam.test_generator.entities.StepResult;
+import com.epam.test_generator.entities.Suit;
+import com.epam.test_generator.entities.SuitResult;
+import com.epam.test_generator.entities.TestResult;
+import com.epam.test_generator.services.exceptions.BadRequestException;
+import com.epam.test_generator.transformers.TestResultTransformer;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Transactional
+@Service
+public class TestResultService {
```

Comment:
Class hard to read, function names are not descriptive. Separating logic to methods did not help a lot.
Remove mixing levels of abstractions
Provide descriptive method names
Provide javadoc
Use recommendations https://herbertograca.com/2016/09/03/clean-code-3-functions-by-robert-c-martin-uncle-bob/ https://sites.google.com/site/unclebobconsultingllc/one-thing-extract-till-you-drop